src/modules/trajectory_generation/domain/spline_generation/cubic_spline_generator.hpp

```diff
@@ -2,19 +2,160 @@
+
+
+            /**
+             * @brief Runs graident descent to compute final spline parameters
+             * 
+             */
+            void run_gradient_descent();
+
+            /**
+             * @brief Outputs a set of postures after computing the final spline params
+             * 
+             * @return std::vector<core_datastructures::Posture> 
+             */
+            std::vector<core_datastructures::Posture> generate_splines();
+
+            /**
+             * @brief Computes curvature given params and arc length
+             * 
+             * @param params 
+             * @param s 
+             * @return double Curvature
+             */
+            double calculate_kappa(std::vector<double>& params, double s);
+
+            /**
+             * @brief Computes theta given params and arc length
+             * 
+             * @param params 
+             * @param s 
+             * @return double theta value in radians
+             */
+            double calculate_theta(std::vector<double>& params, double s);
+
+            // double calculate_x();
+            /**
+             * @brief computes x given params and arc length
+             * 
+             * @param params 
+             * @param s 
+             * @return double 
+             */
+            double calculate_x(std::vector<double>& params, double s);
+
+            /**
+             * @brief Computes y given params and arc length
+             * 
+             * @param params 
+             * @param s 
+             * @return double 
+             */
+            double calculate_y(std::vector<double>& params, double s);
+
+            /**
+             * @brief Gets the euclidian distance given two postures
+             * 
+             * @param start 
+             * @param goal 
+             * @return double eucilidian distance
+             */
+            double get_distance(const core_datastructures::Posture& start, const core_datastructures::Posture& goal);
+
+            
+            /**
+             * @brief return new vector | goal - start |
+             * 
+             * @param start 
+             * @param goal 
+             * @return Eigen::Vector3d 
+             */
+            Eigen::Vector3d diff(const core_datastructures::Posture& start, const core_datastructures::Posture& goal);
+            
+            /**
+             * @brief checks if qiven vector is greater than threhold
+             * 
+             * @param v1  
+             * @return true if values within range of threhold
+             * @return false if values greater than the threshold
+             */
+            bool comp(Eigen::Vector3d& v1);
+
         private:
+            double P0;
+            double P1;
+            double P2;
+            double P3;
+            double SG;
+            
+            /**
+             * @brief Spline convergence threshold value
+             * 
+             */
```

Comment:
No changes required but Its okay to not document private variables. Feel free to resolve.